From: Joel Selvaraj <joelselvaraj.oss@gmail.com>
Date: Sat, 22 Oct 2022 16:26:10 -0500
Subject: arm64: dts: qcom: sdm845-oneplus-common: use separate audio dais

Use separate multimedia for each backend dai as it works more reliably
than sharing same multimedia frontend for multiple backends.

Part-of: <https://gitlab.com/sdm845-mainline/linux/-/merge_requests/53>
---
 .../arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi | 90 ++++++++++++++++++++--
 1 file changed, 84 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi b/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
index 47ed264..300e499 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
@@ -562,6 +562,18 @@ dai@1 {
 	dai@2 {
 		reg = <2>;
 	};
+
+	dai@3 {
+		reg = <3>;
+	};
+
+	dai@4 {
+		reg = <4>;
+	};
+
+	dai@5 {
+		reg = <5>;
+	};
 };
 
 &qupv3_id_1 {
@@ -655,6 +667,27 @@ cpu {
 		};
 	};
 
+	mm4-dai-link {
+		link-name = "MultiMedia4";
+		cpu {
+			sound-dai = <&q6asmdai MSM_FRONTEND_DAI_MULTIMEDIA4>;
+		};
+	};
+
+	mm5-dai-link {
+		link-name = "MultiMedia5";
+		cpu {
+			sound-dai = <&q6asmdai MSM_FRONTEND_DAI_MULTIMEDIA5>;
+		};
+	};
+
+	mm6-dai-link {
+		link-name = "MultiMedia6";
+		cpu {
+			sound-dai = <&q6asmdai MSM_FRONTEND_DAI_MULTIMEDIA6>;
+		};
+	};
+
 	voicemmode1-dai-link {
 		link-name = "VoiceMMode1";
 		cpu {
@@ -673,8 +706,8 @@ platform {
 		};
 	};
 
-	slim-dai-link {
-		link-name = "SLIM Playback";
+	slim1-dai-link {
+		link-name = "SLIM Playback 1";
 		cpu {
 			sound-dai = <&q6afedai SLIMBUS_0_RX>;
 		};
@@ -684,12 +717,12 @@ platform {
 		};
 
 		codec {
-			sound-dai = <&wcd9340 0>;
+			sound-dai = <&wcd9340 0>; /* AIF1_PB */
 		};
 	};
 
-	slimcap-dai-link {
-		link-name = "SLIM Capture";
+	slimcap1-dai-link {
+		link-name = "SLIM Capture 1";
 		cpu {
 			sound-dai = <&q6afedai SLIMBUS_0_TX>;
 		};
@@ -699,7 +732,52 @@ platform {
 		};
 
 		codec {
-			sound-dai = <&wcd9340 1>;
+			sound-dai = <&wcd9340 1>; /* AIF1_CAP */
+		};
+	};
+
+	slim2-dai-link {
+		link-name = "SLIM Playback 2";
+		cpu {
+			sound-dai = <&q6afedai SLIMBUS_1_RX>;
+		};
+
+		platform {
+			sound-dai = <&q6routing>;
+		};
+
+		codec {
+			sound-dai = <&wcd9340 2>; /* AIF2_PB */
+		};
+	};
+
+	slimcap2-dai-link {
+		link-name = "SLIM Capture 2";
+		cpu {
+			sound-dai = <&q6afedai SLIMBUS_1_TX>;
+		};
+
+		platform {
+			sound-dai = <&q6routing>;
+		};
+
+		codec {
+			sound-dai = <&wcd9340 3>; /* AIF2_CAP */
+		};
+	};
+
+	slimcap3-dai-link {
+		link-name = "SLIM Capture 3";
+		cpu {
+			sound-dai = <&q6afedai SLIMBUS_2_TX>;
+		};
+
+		platform {
+			sound-dai = <&q6routing>;
+		};
+
+		codec {
+			sound-dai = <&wcd9340 5>; /* AIF3_CAP */
 		};
 	};
 };
